src/rat/core/run_routing.py

```diff
@@ -47,90 +57,199 @@ def _get_xy(self, lat, lon) -> tuple:
         log.debug("Converted (Lat: %s, Lon: %s ) to (X: %s, Y: %s)", lat, lon, col, row)
         return row, col
     
-    def _read_stations(self, station_path) -> pd.DataFrame:
-        log.debug("Intiializing station DF from: %s", station_path)
-
-        stations = pd.read_csv(station_path)
+    def _read_stations(self, stations) -> pd.DataFrame:
         log.debug(f"{stations}")
         stations['x'] = None
         stations['y'] = None
         stations['filler'] = -9999
         return stations
 
     def create_station_file(self):
+
         if(self.station_file_creation):
             log.debug("Creating station file (X-Y) at: %s", self.station_path_xy)
 
-            self.stations[['y', 'x']] = self.stations.apply(
-                lambda row: pd.Series(
-                        self._get_xy(row['lat'], row['lon']), 
-                        index=['x', 'y']), 
-                        axis=1
-                    )
+            f = lambda row: pd.Series(self._get_xy(row['lat'], row['lon']), index=['x', 'y'])
+
+            self.stations[['y', 'x']] = self.stations.apply(f, axis=1)
             self.stations[['run', 'name', 'x', 'y', 'filler']].to_csv(
-                self.station_path_xy, 
+                self.station_path_xy,
                 sep='\t', 
                 header=False, 
                 index=False, 
-                line_terminator='\nNONE\n'
+                lineterminator='\nNONE\n'
             )
 
-    def run_routing(self):
+    def run_routing(self, cd=None):
         # TODO parse the output of routing model to to remove logs, keep a track of files stations
         #   and their generated output files
         log.log(NOTIFICATION, "Running Routing model")
 
         # args = self.model_path, self.param_path]
-        args = f'cd {self.project_dir} && {self.model_path} {self.param_path}'
-        log.debug("Running: %s", " ".join(args))
-        # ret_code = run_command(['cd',self.project_dir])
-        ret_code = run_command(args,shell=True, executable="/bin/bash")
+        if cd: # if change directory is passed, change directory to that directory
+            # remove any .uh_s files in the directory
+            _ = [f.unlink() for f in Path(cd).glob("*.uh_s")]
+            args = f'{self.model_path} {Path(self.param_path).relative_to(cd)}'
+            log.debug("Running: %s", " ".join(args))
+            log.debug("Changing directory to %s", cd)
+            ret_code = subprocess.run(args.split(), capture_output=True, cwd=cd)
+        else: # else change directory to project directory
+            args = f'{self.model_path} {self.param_path}'
+            log.debug("Running: %s", " ".join(args))
+            ret_code = subprocess.run(args.split(), capture_output=True, shell=True, cwd=self.project_dir)
```

Comment:
Add parameter executable="/bin/bash"

It ensures its applicability to all Linux systems even when by default bash shell is not used.